webapp/src/utils/flattenObject.spec.ts

```diff
@@ -0,0 +1,70 @@
+import flattenObject from './flattenObject';
+import { describe, expect, it } from '@jest/globals';
+
+describe('flattenObject()', () => {
+  it('returns empty object for empty obj', () => {
+    const actual = flattenObject({});
+    expect(actual).toEqual({});
+  });
+  it('flatten obj one string property', () => {
+    const actual = flattenObject({ a: 'b' });
+    expect(actual).toEqual({ a: 'b' });
+  });
+  it('flatten simple obj one object property', () => {
+    const actual = flattenObject({ a: { b: { c: 'd' } } });
+    const expected = { 'a.b.c': 'd' };
+    expect(actual).toEqual(expected);
+  });
+  it('flatten two properties obj', () => {
+    const actual = flattenObject({
+      a: {
+        b: {
+          c: 'd',
+          e: 'f',
+        },
+      },
+    });
+    const expected = {
+      'a.b.c': 'd',
+      'a.b.e': 'f',
+    };
+    expect(actual).toEqual(expected);
+  });
+  it('flatten four properties obj', () => {
+    const actual = {
+      'a.b.c': 'v1',
+      'f.g': 'v2',
+      k: 'v3',
+      // eslint-disable-next-line sort-keys -- so we can test if object were not sorted
```

Comment:
`toEqual` below is not documented to compare key order and the implementation in `@jest/expect-utils/build/jasmineUtils.js` does indeed seem to not compare order.

There is some definitions for iteration order in modern versions of javascript and node seems to support it so it should be possible to test it, but I don't think jest supports it at all. https://node.green/#ES2015-misc-own-property-order

Reply:
removed, there was confusing between Expect and actual